src/material/schematics/ng-generate/mdc-migration/rules/components/list/list-template.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import * as compiler from '@angular/compiler';
+import {TemplateMigrator} from '../../template-migrator';
+import {replaceAttribute, visitElements} from '../../tree-traversal';
+import {Update} from '../../../../../migration-utilities';
+
+export class ListTemplateMigrator extends TemplateMigrator {
```

Comment:
I think for the basic attribute replacements this works perfectly, but what about unscoped content? e.g.

```html
<mat-list-item>
  <span matLine>Title</span>
  At the end
</mat-list-item>
```

should become:

```html
<mat-list-item>
  <span matListItemTitle>Title</span>
  <div matListItemMeta>At the end</div>
</mat-list-item>
```

The internal migration does this. See: `angular2/tools/mdc_migration/template_ast_migrations/list.ts`